pkg/download/classic/download.go

```diff
@@ -279,13 +279,29 @@ func downloadAndUnmarshalConfig(client dtclient.Client, theApi api.API, value va
 		return nil, err
 	}
 
-	if values, ok := data[theApi.PropertyNameOfGetAllResponse]; ok {
-		var res []map[string]any
-		err := mapstructure.Decode(values, &res)
-		return res, err
+	values, found := data[theApi.PropertyNameOfGetAllResponse]
+	if !found {
+		return []map[string]any{data}, nil
 	}
 
-	return []map[string]any{data}, nil
+	var res []map[string]any
+	err = mapstructure.Decode(values, &res)
+	if err != nil {
+		return []map[string]any{}, err
+	}
+	if theApi.ID == api.KeyUserActionsWeb {
```

Comment:
whenever we add special treatment with an if like that, please explain why with a short comment

Reply:
i mean it is not clear why we need to inspect the payload and do all of that for that config type ;)